spec/controllers/api/irs_attempts_api_controller_spec.rb

```diff
@@ -107,6 +79,72 @@
         expect(Base64.strict_decode64(response.headers['X-Payload-Key'])).to be_present
         expect(response.body).to eq(test_object)
       end
+
+      context 'with aws_s3_stream enabled' do
+        let(:test_object) { '{test: "1234567890 12345"}' }
+        before do
+          allow(IdentityConfig.store).to receive(:irs_attempt_api_aws_s3_stream_enabled).
+            and_return(true)
+          allow(IdentityConfig.store).to receive(:irs_attempt_api_aws_s3_stream_buffer_size).
+            and_return(10)
+
+          Aws.config[:s3] = {
+            stub_responses: {
+              head_object: { content_length: test_object.bytesize },
+              get_object: proc do |context|
+                range_string = context.params[:range]
+                byte_string = range_string.split('=')[1]
+                byte_range_array = byte_string.split('-')
+                { body: test_object.byteslice(
+                  byte_range_array[0].to_i,
+                  IdentityConfig.store.irs_attempt_api_aws_s3_stream_buffer_size + 1,
+                ) }
+              end,
+            },
+          }
+        end
+
+        it 'should render data streamed from s3 correctly' do
+          post :create, params: { timestamp: time.iso8601 }
+
+          expect(response).to be_ok
+          expect(Base64.strict_decode64(response.headers['X-Payload-IV'])).to be_present
+          expect(Base64.strict_decode64(response.headers['X-Payload-Key'])).to be_present
+          expect(response.content_type).to eq('application/octet-stream')
+          expect(response['Content-Disposition']).
+            to eq("attachment; filename=\"test_filename\"; filename*=UTF-8''test_filename")
```

Comment:
Note to self: This weird `filename*=UTF-8` business is apparently [RFC-5987](https://datatracker.ietf.org/doc/html/rfc5987#section-3.2.2).